rosmon_core/src/launch/launch_config.cpp

```diff
@@ -381,26 +387,30 @@ void LaunchConfig::parseNode(TiXmlElement* element, ParseContext ctx)
 	if(!fullNamespace.empty())
 		node->setNamespace(fullNamespace);
 
-	if(respawn)
+	if (m_respawnObey && respawn)
 	{
 		node->setRespawn(ctx.parseBool(respawn, element->Row()));
-
-		if(respawnDelay)
+	} 
+	else 
+	{
+		node->setRespawn(m_respawnAll);
+	}
+        
+	if(((m_respawnObey && respawn && ctx.parseBool(respawn, element->Row())) || m_respawnAll) && respawnDelay)
```

Comment:
I think instead of complicated chain of conditions here, the only change needed in this function would be to swap the following in original implementation 
```
if (respawn)
{
    node->setRespawn(ctx.parseBool(respawn, element->Row()));
```
with
```
if ((m_respawnObey && respawn) || !m_respawnObey)
{
   if (!m_respawnObey) {
        node->setRespawn(m_respawnAll);
    } else {
        node->setRespawn(ctx.parseBool(respawn, element->Row()));
    }
```

Reply:
I don't think this would work because then respawn delay will be set even if m_respawnAll is set to false.